xpdan/search.py

```diff
@@ -0,0 +1,190 @@
+from pyxdameraulevenshtein import \
+    normalized_damerau_levenshtein_distance as ndld
+from databroker.broker import _munge_time
+from pprint import pprint
+import pytz
+from heapq import heapify, heappushpop
+
+
+def get_from_dict(dataDict, mapList):
+    """ Get a value from a nested dictionary, given a list of keys
+
+    Parameters
+    ----------
+    dataDict: dict
+        The dictionary to be queried
+    mapList: list of str
+        A list of strings, each string is one level lower than the previous
+
+    Returns
+    -------
+    object:
+        The the value from the dict
+
+    """
+    for k in mapList:
+        dataDict = dataDict[k]
+    return dataDict
+
+
+def nested_dict_values(d):
+    """Yield all string values inside a nested dictionary
+
+    Parameters
+    ----------
+    d: dict
+        The dictionary to be unpacked
+
+    Yields
+    -------
+    str:
+        The string value inside the dictionary
+
+    """
+    for v in d.values():
+        if isinstance(v, dict):
+            yield from nested_dict_values(v)
+        else:
+            if isinstance(v, str):
+                yield v
+            else:
+                yield None
+
+
+def fuzzy_search(db, keys, search_string, size=100):
```

Comment:
Refactor this to take in kwargs as searches
```python
hdrs = fuzzy_search(db, piLast='foo')
```
This is much closer to databroker search syntax. Actually we might want to mirror it entirely with the exception of having a db arg.